File: backend/LibraryApi.Domain/src/Entities/Loan.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryApi.Domain.src.Entities;

public class Loan : BaseEntity
{
  public bool IsReturned { get; set; }
  public Guid UserId { get; set; }
  public Guid BookId { get; set; }
  public User User { get; set; } = null!;
  public Book Book { get; set; } = null!;
  // public DateOnly DueDate { get; set; }  
}

File: frontend/src/types/Loan.ts
import { Book } from "./Book"
import { User } from "./User"

export interface Loan {
  id: string
  bookId: string
  userId: string
  isReturned: boolean
}

export interface LoanReadDto {
  id: string
  book: Book
  user: User
  isReturned: boolean
}